app/controllers/application_controller.rb

```diff
@@ -1,2 +1,42 @@
 class ApplicationController < ActionController::API
+  attr_reader :current_user, :token
+
+  def no_route_found
+    found = { Error: "The end point you requested does not exist.",
+              Debug: "Please check the documentation for existing end points" }
+    render json: found, status: 404
+  end
+
+  def authenticate
+    @token = request.headers["HTTP_AUTHORIZATION"]
+    unless token_has_expired(token)
+      status, payload = Api::V1::Authenticate.decode_token(token)
```

Comment:
I think since they are in the same namespace we can call `Authenticate.decode_token)` directly